app/queues/dns/workers/dns-update-worker.server.ts

```diff
@@ -0,0 +1,49 @@
+import { UnrecoverableError, Worker } from 'bullmq';
+
+import { redis } from '~/lib/redis.server';
+import logger from '~/lib/logger.server';
+import { createRecord, upsertRecord, deleteRecord } from '~/lib/dns.server';
+
+import type { WorkType } from '../add-record-flow.server';
+import type { Record } from '@prisma/client';
+
+export const dnsUpdateQueueName = 'dns-record-update';
+
+export interface DnsUpdaterData {
+  workType: WorkType;
+  username: Record['username'];
+  type: Record['type'];
+  fqdn: string;
```

Comment:
If we rely on subdomain in the whole process, let's use that here too, and generate fqdn from the username and subdomain here. Should simplify the flow creator a bit